File: frontend/src/services/reservationService.ts
import api from './api';
import { Reservation, ReservationRequest } from '../types';

export const reservationService = {
  async createReservation(data: ReservationRequest): Promise<Reservation> {
    const response = await api.post<Reservation>('/reservations', data);
    return response.data;
  },

  async getReservationById(id: number): Promise<Reservation> {
    const response = await api.get<Reservation>(`/reservations/${id}`);
    return response.data;
  },

  async getMyReservations(): Promise<Reservation[]> {
    const response = await api.get<Reservation[]>('/reservations/my-reservations');
    return response.data;
  },

  async cancelReservation(id: number): Promise<Reservation> {
    const response = await api.post<Reservation>(`/reservations/${id}/cancel`);
    return response.data;
  },
};

File: backend/src/main/java/com/hotel/management/service/ReservationService.java
package com.hotel.management.service;

import com.hotel.management.dto.request.ReservationRequest;
import com.hotel.management.dto.response.ReservationResponse;
import com.hotel.management.entity.User;

import java.util.List;

public interface ReservationService {
    ReservationResponse createReservation(ReservationRequest request, User user);
    ReservationResponse getReservationById(Long id);
    List<ReservationResponse> getMyReservations(User user);
    ReservationResponse cancelReservation(Long id, User user);
    ReservationResponse checkIn(Long id);
    ReservationResponse checkOut(Long id);
}
